test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatLocalsTests.cs

```diff
@@ -26,19 +29,113 @@ public class XPlatLocalsTests
         [InlineData("locals global-packages -l")]
         [InlineData("locals --list global-packages")]
         [InlineData("locals -l global-packages")]
-        // [InlineData("locals --clear all")]
-        // [InlineData("locals -c all")]
-        public static void Locals_Succeeds(String args)
+        public static void Locals_List_Succeeds(string args)
         {
-            // Arrange
-            var log = new TestCommandOutputLogger();
+            using (var mockBaseDirectory = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange
+                var mockGlobalPackagesDirectory = Directory.CreateDirectory(Path.Combine(mockBaseDirectory.Path, @"global-packages"));
+                var mockHttpCacheDirectory = Directory.CreateDirectory(Path.Combine(mockBaseDirectory.Path, @"http-cache"));
+                var mockTmpCacheDirectory = Directory.CreateDirectory(Path.Combine(mockBaseDirectory.Path, @"temp"));
 
-            // Act
-            var exitCode = Program.MainInternal(args.Split(null), log);
+                DotnetCliUtil.CreateTestFiles(mockGlobalPackagesDirectory.FullName);
+                DotnetCliUtil.CreateTestFiles(mockHttpCacheDirectory.FullName);
+                DotnetCliUtil.CreateTestFiles(mockTmpCacheDirectory.FullName);
 
-            // Assert
-            Assert.Equal(string.Empty, log.ShowErrors());
-            Assert.Equal(0, exitCode);
+                // Act
+                var result = CommandRunner.Run(
```

Comment:
What happens if the running process does not have access to delete a file or directory in one of these locals directories? Does it fail in the middle of the recursive delete? Does it skip that inaccessible file?